src/Internal/Search/Searcher.php

```diff
@@ -346,6 +362,50 @@ private function addTermDocumentMatchesCTEs(TokenCollection $tokenCollection): v
         }
     }
 
+    private function addTermDocumentsCTE(Token $token): void
+    {
+        // No term matches CTE -> no term documents CTE
+        $termMatchesCTE = $this->getCTENameForToken(self::CTE_TERM_MATCHES_PREFIX, $token);
+
+        if (!$this->hasCTE($termMatchesCTE)) {
+            return;
+        }
+
+        $termsDocumentsAlias = $this->engine->getIndexInfo()->getAliasForTable(IndexInfo::TABLE_NAME_TERMS_DOCUMENTS);
+
+        $cteSelectQb = $this->engine->getConnection()->createQueryBuilder();
+        $cteSelectQb->addSelect('DISTINCT ' . $termsDocumentsAlias . '.document');
```

Comment:
```suggestion
        $cteSelectQb->select($termsDocumentsAlias . '.document')->distinct()
```

is what I have used elsewhere :)

Reply:
Makes sense!